### PR TITLE
sstable: add range keys to sstable.Layout

### PR DESCRIPTION
**Pull request by nicktrav**
The `sstable.Layout` struct contains information pertaining to the
layout of an sstable. Add the range key block to the layout.

Related to #1339.

### CODE REVIEW

File: sstable/properties.go [hunk 1]
```diff
@@ -133,9 +133,9 @@ type Properties struct {
 	// Total raw key size.
 	RawKeySize uint64 `prop:"rocksdb.raw.key.size"`
 	// Total raw rangekey key size.
-	RawRangeKeyKeySize uint64 `prop:"pebble.raw.rangekey.key.size"`
+	RawRangeKeyKeySize uint64 `prop:"pebble.raw.range-key.key.size"`
```
Comment (nicktrav) [inline sstable/properties.go:136]: These two changes are purely cosmetic, and make the range key property names consistent (i.e. hyphenated).

**Review by jbowens — APPROVED** [review]
:LGTM:

Seeing this PR reminded me of one more range-key block change that we should consider doing: the `*sstable.Reader` type has an `EstimateDiskUsage` method. If the sstable has range keys, I think we should check if the range key block contains any range keys that overlap with the key span passed to `EstimateDiskUsage`. If there are, we should add the entire length of the range-key block to the estimate returned.

> Reviewed 2 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/pebble/1410#-:-MqzlLbG5yoL3EJaybfv:b-58byup)* status: 2 of 3 files reviewed, 1 unresolved discussion (waiting on @sumeerbhola)

**Review by sumeerbhola — APPROVED** [review]
:lgtm: 

> Reviewed 2 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/pebble/1410#-:-Mr3CDdNCail9Yvw_zJi:bg7igkv)* status: all files reviewed, 1 unresolved discussion (waiting on @jbowens and @sumeerbhola)

**Review by nicktrav — COMMENTED** [review]
> If there are, we should add the entire length of the range-key block to the estimate returned.

Ack. Will ensure we're tracking this task.

TFTRs!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/pebble/1410#-:-Mr3IlQI7RKjFo7fb4f7:ba9swxa)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @nicktrav)

**Pull request merged**